rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java

```diff
@@ -113,6 +114,29 @@ public void putBlocks(long blockNumber, List<IndexedBlockStore.BlockInfo> blocks
         index.put(blockNumber, blocks);
     }
 
+    @Override
+    public void removeBlock(long blockNumber, Keccak256 blockHash) {
+        List<IndexedBlockStore.BlockInfo> blockInfoList = index.get(blockNumber);
+
+        if (blockInfoList == null) {
+            return;
+        }
+
+        List<IndexedBlockStore.BlockInfo> toRemove = new ArrayList<>();
+
+        for (IndexedBlockStore.BlockInfo bInfo : blockInfoList) {
+            if (bInfo.getHash().equals(blockHash)) {
+                toRemove.add(bInfo);
+            }
+        }
+        blockInfoList.removeAll(toRemove);
+        if (blockInfoList.isEmpty()) {
+            index.remove(blockNumber);
+        } else {
+            index.put(blockNumber, blockInfoList);
```

Comment:
Maybe we can add a comment explaining why this is needed, so the next dev working here does not try to change it.